src/worker.rs

```diff
@@ -83,23 +94,59 @@ async fn start_worker(server_addr: String) {
     };
 
     let fut = async move {
-        let endpoint = match Endpoint::from_shared(server_addr) {
-            Ok(endpoint) => endpoint,
+        let (tx, rx) = mpsc::channel::<Result<CollectItem, Box<dyn Error + Send>>>(255);
+        let listener = match UnixListener::bind(worker_addr) {
+            Ok(listener) => listener,
             Err(err) => {
-                error!(?err, "Create endpoint failed");
+                error!(?err, "Bind failed");
                 return;
             }
         };
-        let channel = connect(endpoint).await;
 
-        let consumer = match channel::Consumer::new() {
-            Ok(consumer) => consumer,
+        debug!("Bind");
+        tokio::spawn(async move {
+            loop {
+                match listener.accept().await {
+                    Ok((mut stream, _addr)) => {
+                        debug!("Accept");
+
+                        let tx = tx.clone();
+                        tokio::spawn(async move {
+                            loop {
+                                let r = match channel::channel_receive(&mut stream).await {
+                                    Err(err) => match err.downcast_ref::<io::Error>() {
+                                        Some(e) if e.kind() == io::ErrorKind::UnexpectedEof => {
+                                            return
+                                        }
+                                        _ => Err(err.into()),
+                                    },
+                                    Ok(i) => Ok(i),
+                                };
+
+                                if let Err(err) = tx.send(r).await {
+                                    error!(?err, "Send failed");
+                                    return;
+                                }
+                            }
+                        });
+                    }
```

Comment:
It seems that `(tx, rx)` and outer `tokio::spawn` can be removed, to prevent the data copied. In addition, please write more details in the log, like `debug!(server_addr, "Bind unix domain socket");`.

Reply:
Without an outer `spawn`, the `accept` (dead) loop will block the codes after it. Perhaps we could reorg the codes into another `Future` and prevent an explicit spawn (but an arm of `select!`). I'm not sure whether this is necessary.
The `mpsc` facilitates `Consumer`. If we build the worker listener with a `select` structure, we may be able to replace the `mpsc` with a FIFO. Still, the `UnixListener` and the `GrpcReporter` seem to run in their own (tokio) threads. IMHO, (tokio) `mpsc` is proper here, at least with the current schema.